portal-kernel/src/com/liferay/portal/kernel/util/PortalFunction.java

```diff
@@ -0,0 +1,18 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.kernel.util;
+
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author Lily Chi
+ */
+@FunctionalInterface
+public interface PortalFunction<T, R> {
```

Comment:
@slnn `UnsafeFunction<String, Company, PortalException>` is enough for this. But see my other comment, we'd better not use any function at all.